### ISSUE TITLE
showing blank jpeg image.

### ISSUE THREAD

**Issue by raginisahu** [nicoFuccella/pdf-thumbnail#4]
const pdf = require('pdf-thumbnail');
      const pdfBuffer = require('fs').readFileSync('/home/ragini/Desktop/front page.pdf');
 
pdf(
  pdfBuffer, /*Buffer or stream of the pdf*/
  {
    crop: {
      width: 200,
      height: 400,
      x: 0,
      y: 0
    }
  })
  .then(data /*Stream of the image*/ => data.pipe(fs.createWriteStream(join('/home/ragini/Desktop', "preview", "previewBuffer.jpg"))))
    // ...

please give me solution.

**Comment by nicofuccella** [comment]
@raginisahu could you give me more info about the issue?

**Comment by AmirHossein** [comment]
I have same problem. It works without error, but length of `data` (on `then`) is 0 and it creates a image file with zero bytes.

**Comment by nicofuccella** [comment]
@AmirHossein @raginisahu i need more info about the operating system you are currently using and other global settings

**Comment by iamcrisb** [comment]
Encountered the same problem using Ubuntu 18.04 subsystem for Windows running it with node v9.8.0 npm version 5.6.0

**Comment by raginisahu** [comment]
I am using Mac operating system with node js 11.13.0 .

**Comment by Ng2k** [comment]
@raginisahu @cristianbuta @AmirHossein have you already installed imagemagick and ghostscript successfully?

**Comment by iamcrisb** [comment]
@Ng2k yes. Had no problems with the installation. Also used some different modules that had imagemagick as a dependency and it worked perfectly.

**Comment by Ng2k** [comment]
@cristianbuta @raginisahu @AmirHossein I've just found [this post](https://stackoverflow.com/a/52661288) on stackoverflow. Here they explain how ImageMagick doesn't give permission to Ghostscript to manipulate pdf files, because there are security problems. You have to modify the Imagemagick policy.xml in order to read and write pdf files.

**IMPORTANT:** read carefully the security problems of ghostscript.

I hope this comment could be helpful for you. Alert us for everything else.

**Comment by yuvarajvemula** [comment]
I am using ubuntu 8.04 .. getting no error in promise.. but only getting empty image

**Comment by nullromo** [comment]
@Ng2k's suggestion based on the Stack Overflow link works for me, and I only had to change 1 line in `/etc/ImageMagick-6/policy.xml`

```diff
-    <policy domain="coder" rights="none" pattern="PDF" />
+    <policy domain="coder" rights="read|write" pattern="PDF" />
```

This worked on Windows Subsystem for Linux with Node v12.18.0

**Comment by minhna** [comment]
I confirm the @nullromo solution works on Ubuntu 20.04 with node version v14.18.3

**Comment by redeemefy** [comment]
My guess is that if the pdf is password protected this library can't generate the thumbnail. Is that correct?

I still wonder how google can generate a preview of the 1st page or a password protected pdf and use it as thumbnail in the drive application. If one uploads a pdf to google drive, google is able to generate a thumbnail even though the pdf is password protected.

**Comment by RiedelNicolas** [comment]
You have to change what @nullromo said.

Here's the sed command you can use if you're working in an automated environment :

```sh
sed -i 's/<policy domain="coder" rights="none" pattern="PDF" \/>/<policy domain="coder" rights="read|write" pattern="PDF" \/>/' /etc/ImageMagick-6/policy.xml
```
